perses/app/relative_point_mutation_setup.py

```diff
@@ -159,23 +159,31 @@ def __init__(self,
 
         # Load the ligand, if present
         molecules = []
-        if ligand_file:
-            if ligand_file.endswith('.sdf'): # small molecule
-                ligand_mol = createOEMolFromSDF(ligand_file, index=ligand_index)
-                ligand_mol = generate_unique_atom_names(ligand_mol)
+        if ligand_input:
+            if isinstance(ligand_input, str):
+                if ligand_input.endswith('.sdf'): # small molecule
+                        ligand_mol = createOEMolFromSDF(ligand_input, index=ligand_index)
+                        ligand_mol = generate_unique_atom_names(ligand_mol)
+                        molecules.append(Molecule.from_openeye(ligand_mol, allow_undefined_stereo=False))
+                        ligand_positions, ligand_topology = extractPositionsFromOEMol(ligand_mol),  forcefield_generators.generateTopologyFromOEMol(ligand_mol)
+                        ligand_md_topology = md.Topology.from_openmm(ligand_topology)
+                        ligand_n_atoms = ligand_md_topology.n_atoms
+
+                if ligand_input.endswith('pdb'): # protein
+                    ligand_pdbfile = open(ligand_input, 'r')
+                    ligand_pdb = app.PDBFile(ligand_pdbfile)
+                    ligand_pdbfile.close()
+                    ligand_positions, ligand_topology, ligand_md_topology = ligand_pdb.positions, ligand_pdb.topology, md.Topology.from_openmm(
+                        ligand_pdb.topology)
+                    ligand_n_atoms = ligand_md_topology.n_atoms
+
+            elif isinstance(ligand_input, list): # list of oemol object(s)
+                ligand_mol = ligand_input[0] # assume only one molecule in the list
                 molecules.append(Molecule.from_openeye(ligand_mol, allow_undefined_stereo=False))
```

Comment:
I think `ligand_mol = generate_unique_atom_names(ligand_mol)` is necessary to include before this line.. did you see any errors when you ran with this version of `PointMutationExecutor`?

Reply:
oh wait, i just noticed `generate_unique_atom_names` is redundant. 
it's called here: https://github.com/choderalab/perses/blob/f640bf82d75262530e68255f4d4d5fc43f70eea3/perses/utils/openeye.py#L354
so it doesn't need to be called again after `createMolFromSDF`